api/v1alpha1/common_types.go

```diff
@@ -0,0 +1,31 @@
+package v1alpha1
+
+import corev1 "k8s.io/api/core/v1"
+
+// Autoscaling configuration for scalable workloads.
+type Autoscaling struct {
+	// MaxReplicas is the upper limit for the number of replicas to which the autoscaler can scale up.
+	// It cannot be less than the replica count for your workload.
+	MaxReplicas int32 `json:"maxReplicas"`
+
+	// AverageUtilization is the target value of the average of the resource metric across all relevant pods.
+	// This is represented as a percentage of the requested value of the resource for the pods.
+	AverageUtilization int32 `json:"averageUtilization"`
+}
+
+// OCIImageDefinition describes where and when to fetch a container image.
+type OCIImageDefinition struct {
+	// Registry where the container image is hosted.
+	// +kubebuilder:validation:Optional
+	Registry string `json:"registry"`
+
+	// Repository where the container image is stored.
+	Repository string `json:"repository"`
+
+	// Tag points to a specific container image variant.
+	Tag string `json:"tag"`
+
+	// PullPolicy used to fetch container image.
+	// +kubebuilder:validation:Optional
+	PullPolicy corev1.PullPolicy `json:"pullPolicy"`
+}
```

Comment:
Does this need a pull secret?

Reply:
The only thing worth note here is that I created the pull secret field directly on the spec as a slice of ref objects. This way N number of pull secrets can be added to pull both the ray image and any additional init container images if/when they live in different private registries.